pkg/hintrunner/hinter/zero_dict.go

```diff
@@ -83,36 +83,46 @@ func (dm *ZeroDictionaryManager) NewDefaultDictionary(vm *VM.VirtualMachine, def
 // Given a memory address, it looks for the right dictionary using the segment index. If no
 // segment is associated with the given segment index, it errors
 func (dm *ZeroDictionaryManager) GetDictionary(dictAddr mem.MemoryAddress) (ZeroDictionary, error) {
-	dict, ok := dm.dictionaries[dictAddr.SegmentIndex]
-	if ok {
-		return dict, nil
+	dict, ok := dm.Dictionaries[dictAddr.SegmentIndex]
+	if !ok {
+		return ZeroDictionary{}, fmt.Errorf("no dictionary at address: %s", dictAddr)
 	}
-	return ZeroDictionary{}, fmt.Errorf("no dictionary at address: %s", dictAddr)
+	if *dict.FreeOffset != dictAddr.Offset {
+		return ZeroDictionary{}, fmt.Errorf("no dictionary at address: %s", dictAddr)
+	}
+	return dict, nil
 }
 
 // Given a memory address and a key it returns the value held at that position. The address is used
 // to locate the correct dictionary and the key to index on it
 func (dm *ZeroDictionaryManager) At(dictAddr mem.MemoryAddress, key f.Element) (mem.MemoryValue, error) {
-	if dict, ok := dm.dictionaries[dictAddr.SegmentIndex]; ok {
-		return dict.At(key)
+	dict, err := dm.GetDictionary(dictAddr)
+	if err != nil {
+		return mem.UnknownValue, err
+	}
+	value, err := dict.At(key)
```

Comment:
i think just made dict methods private in a previous PR, does it work like this?

Reply:
oh yeah its just conflicts to resolve

Reply:
fixed conflicts!